File: Java/src/test/java/us/fatehi/palindrome/PalindromeTest.java
package us.fatehi.palindrome;


import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class PalindromeTest
{

  private final Palindrome p = new Palindrome();

  @Test
  public void edgeCaseNull()
  {
    assertTrue(p.isPalindrome(null));
  }

  @Test
  public void edgeCaseEmptyString()
  {
    assertTrue(p.isPalindrome(""));
  }

  @Test
  public void happyPathSingleLetter()
  {
    assertTrue(p.isPalindrome("a"));
  }

  @Test
  public void happyPathEvenLetters()
  {
    assertTrue(p.isPalindrome("noon"));
  }

  @Test
  public void happyPathOddLetters()
  {
    assertTrue(p.isPalindrome("racecar"));
  }

  @Test
  public void negativeCase()
  {
    assertFalse(p.isPalindrome("burning"));
  }

}

File: C#/TestPalindrome/PalindromeTest.cs
using NUnit.Framework;
using us.fatehi.palindrome;

namespace us.fatehi.palindrome.test
{

    [TestFixture]
    public class PalindromeTest
    {
        private readonly Palindrome _p = new Palindrome();

        [Test]
        public void EdgeCaseNull()
        {
            Assert.That(_p.IsPalindrome(null));
        }

        [Test]
        public void EdgeCaseEmptyString()
        {
            Assert.That(_p.IsPalindrome(""));
        }

        [Test]
        public void HappyPathSingleLetter()
        {
            Assert.That(_p.IsPalindrome("a"));
        }

        [Test]
        public void HappyPathEvenLetters()
        {
            Assert.That(_p.IsPalindrome("noon"));
        }

        [Test]
        public void HappyPathOddLetters()
        {
            Assert.That(_p.IsPalindrome("racecar"));
        }

        [Test]
        public void NegativeCase()
        {
            Assert.That(_p.IsPalindrome("burning"), Is.False);
        }
    }
}